ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/AsyncSupport.scala

```diff
@@ -29,34 +30,24 @@ object AsyncSupport {
   def asyncPool(
       size: Int,
       namePrefix: String,
-      withMetric: Option[(MetricName, MetricRegistry)] = None,
+      withMetric: (MetricName, MetricRegistry, ExecutorServiceMetrics),
   )(implicit loggingContext: LoggingContext): ResourceOwner[Executor] =
     ResourceOwner
-      .forExecutorService(() =>
-        ExecutionContext.fromExecutorService(
-          {
-            val executor = Executors.newFixedThreadPool(
-              size,
-              new ThreadFactoryBuilder()
-                .setNameFormat(s"$namePrefix-%d")
-                .build,
-            )
-            withMetric match {
-              case Some((metricName, metricRegistry)) =>
-                new InstrumentedExecutorService(
-                  executor,
-                  metricRegistry,
-                  metricName,
-                )
-
-              case None => executor
-            }
-          },
+      .forExecutorService { () =>
+        val (executorName, metricRegistry, executorServiceMetrics) = withMetric
+        executors.InstrumentedExecutors.newFixedThreadPoolWithFactory(
```

Comment:
Just wondering, why not reference `InstrumentedExecutors` directly here (i.e., without `executors` package prefix)?

Reply:
Will remove that, it was because of conflicts before the rename.